lightkurve/correctors/pldcorrector.py

```diff
@@ -85,52 +84,151 @@ class PLDCorrector(Corrector):
+
+        Returns
+        -------
+        dm : `.DesignMatrixCollection`
+            `.DesignMatrixCollection` containing pixel, background, and spline
+            components.
+        """
+
+        if background_mask is None:
+            # Default to pixels <1-sigma above the background
+            background_mask = ~self.tpf.create_threshold_mask(1, reference_pixel=None)
+        self.background_mask = background_mask
+
+        DMC, spline = DesignMatrixCollection, create_spline_matrix
+        if sparse:
+            DMC, spline = SparseDesignMatrixCollection, create_sparse_spline_matrix
+        # First, we estimate the per-pixel background flux over time by
+        # (i) subtracting a mean image from each cadence;
+        # (ii) computing the median pixel value in the residual images;
+        # (iii) assume that the 5%-percentile of those medians gives us the
+        # exact background level. This assumption appears to work well for TESS
+        # but it has not been validated in detail yet.
+        simple_bkg = (self.tpf.flux - np.nanmean(self.tpf.flux, axis=0))
+        simple_bkg = np.nanmedian(simple_bkg[:, background_mask], axis=1)
+        simple_bkg -= np.percentile(simple_bkg, 5)
+
+        # Parse PLD aperture mask
+        pld_pixel_mask = self.tpf._parse_aperture_mask(pld_aperture_mask)
+
+        # Background-subtracted, flux-normalized pixel time series
+        regressors = self.tpf.flux[:, pld_pixel_mask].reshape(len(self.tpf.flux), -1)
+        regressors = regressors - simple_bkg.reshape(-1,1)
+        regressors = np.array([r[np.isfinite(r)] for r in regressors])
+        regressors = np.array([r / f for r,f in zip(regressors, self.lc.flux.value)])
+
+        # Create first order design matrix
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore')
+            pld_1 = DesignMatrix(regressors).pca(pixel_components)
+
+        # Create higher order matrix
+        all_pld = [pld_1]
+        for i in range(2, pld_order+1):
+            # This step creates higher order products of pixel components, from 2nd to nth order
+            reg_n = np.product(list(multichoose(pld_1.values.T, i)), axis=1).T
+            # Apply PCA before merging into single PLD matrix
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore')
+                pld_n = DesignMatrix(reg_n).pca(pixel_components)
+            all_pld.append(pld_n)
+
+        # Collect each matrix
+        dm_pixels = DesignMatrixCollection(all_pld).to_designmatrix(name='pixel_series')
+        dm_bkg = DesignMatrix(simple_bkg, name='background_model')
+        dm_spline = spline(self.lc.time.value, n_knots=spline_n_knots,
+                             degree=spline_degree).append_constant()
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore')
+            dm = DMC([dm_pixels, dm_bkg, dm_spline])
+
+        self.dm = dm
```

Comment:
I recommend removing this to avoid confusion, because `RegressionCorrector.correct()` already stores the design matrix as `self.design_matrix_collection`:
```suggestion
        self.dm = dm
```